packages/orbit-components/src/primitives/ButtonPrimitive/index.tsx

```diff
@@ -1,263 +1,283 @@
+          varsButton["--button-box-shadow-focus"] != null &&
+            "focus:[box-shadow:var(--button-box-shadow-focus)]",
+          varsButton["--button-box-shadow-active"] != null &&
+            "active:[box-shadow:var(--button-box-shadow-active)] active:focus:[box-shadow:var(--button-box-shadow-active)]",
+        )}
+        style={varsButton}
+      >
         {loading && <Loading type="buttonLoader" />}
-        <ButtonPrimitiveContent loading={loading} contentAlign={contentAlign}>
-          {iconLeft && (
-            <ButtonPrimitiveIconContainer width={width} height={height} margin={leftMargin}>
-              {iconLeft}
-            </ButtonPrimitiveIconContainer>
-          )}
-          {children && (
-            <ButtonPrimitiveContentChildren
-              hasIcon={Boolean(iconLeft || iconRight)}
-              contentWidth={contentWidth}
-              centered={fullWidth && centered}
-            >
-              {children}
-            </ButtonPrimitiveContentChildren>
-          )}
-          {iconRight && (
-            <ButtonPrimitiveIconContainer width={width} height={height} margin={rightMargin}>
-              {iconRight}
-            </ButtonPrimitiveIconContainer>
-          )}
-        </ButtonPrimitiveContent>
-      </StyledButtonPrimitive>
+
+        {iconLeft != null && (
+          <div
+            className={cx(
+              "orbit-button-primitive-icon flex items-center justify-center",
+              loading === true && "invisible",
+              // icon vars
+              varsIcons["--button-icon-height"] != null && "[&>svg]:h-[var(--button-icon-height)]",
+              varsIcons["--button-icon-width"] != null && "[&>svg]:w-[var(--button-icon-width)]",
+              varsIcons["--button-icon-left-margin"] != null &&
+                children != null &&
+                "m-[var(--button-icon-left-margin)]",
+              varsIcons["--button-icon-foreground"] != null &&
+                "text-[color:var(--button-icon-foreground)] group-disabled:text-[color:var(--button-icon-foreground)]",
+              varsIcons["--button-icon-foreground-hover"] != null &&
+                "group-hover:text-[color:var(--button-icon-foreground)]",
+              varsIcons["--button-icon-foreground-focus"] != null &&
+                "group-focus:text-[color:var(--button-icon-foreground)]",
+              varsIcons["--button-icon-foreground-active"] != null &&
+                "group-active:text-[color:var(--button-icon-foreground)] group-active:group-focus:text-[color:var(--button-icon-foreground)]",
+            )}
+            style={varsIcons}
+          >
+            {iconLeft}
+          </div>
+        )}
+        {children != null && (
+          <div
+            className={cx(
+              "orbit-button-primitive-content inline-block",
+              (iconLeft != null || iconRight != null) && "text-start",
+              loading === true && "invisible",
```

Comment:
I can understand usage of *!= null* or *== null* equality checks 

but for such cases with true, I would prefer just to have *&&* 

I do not mind having all of these, but personally, I found it harder to read and the benefit of it is questionable, there are cases when it's especially useful, but dunno if it's a good idea to promote it everywhere. :)

Reply:
I am used to having [strict boolean expressions(https://typescript-eslint.io/rules/strict-boolean-expressions/) in place and use the code style even when the rule is not enabled. it imo has a lot of explicitness value and I'd argue it makes things _more_ readable, because you don't have to second-guess types of expressions

Reply:
later we can think about unifying this, since you already refactored several components and the code % with this will be increasing